src/tracer.rs

```diff
@@ -101,9 +98,36 @@ impl ObjectSubclass for TracingTracerPriv {
         }
     }
 }
+pub(crate) trait TracingTracerImpl: TracerImpl {}
+
+unsafe impl<T: TracingTracerImpl> IsSubclassable<T> for TracingTracer {
+    fn class_init(class: &mut glib::Class<Self>) {
+        Self::parent_class_init::<T>(class);
+    }
+}
 
 impl ObjectImpl for TracingTracerPriv {
     fn constructed(&self) {
+        if let Some(params) = self.obj().property::<Option<String>>("params") {
+            let tmp = format!("params,{}", params);
+            info!("{:?} params: {:?}", self.obj(), tmp);
+            let structure = gstreamer::Structure::from_str(&tmp).unwrap_or_else(|e| {
+                error!("Invalid params string: {:?}: {e:?}", tmp);
+                gstreamer::Structure::new_empty("params")
+            });
+
+            if let Ok(gst_logs_level) = structure
+                .get::<String>("log-level")
+                .or_else(|_| structure.get::<i32>("log-level").map(|l| l.to_string()))
+            {
+                info!("Integrating `{gst_logs_level}` GStreamer logs as part of our tracing");
+
+                crate::integrate_events();
+                gstreamer::debug_remove_default_log_function();
+                gstreamer::debug_set_threshold_from_string(&gst_logs_level, true);
```

Comment:
These seem like they should be a part of the subobjects that implement specific output strategies. Users who want to integrate this library directly could decide that they want to e.g. keep both the default gstreamer log function…

Reply:
I think any subclass should have the feature, this is behind the `log-level` parameter the can activate it "manually" if using as library. Also right now there not really a way to pass those parameters when using as a library.